docs/user_guide/problem_parallelization.md

```diff
@@ -34,23 +34,36 @@ By using Ray for parallelisation by default, EvoTorch therefore supports deploym
 
 In our guide on [Defining Problems](problems.md), we demonstrated the use of CUDA-capable devices using the `device` argument. However, as Ray communicates between actors on the CPU, it is recommended that when you have `num_actors > 1`, you use the default value `device = 'cpu'` so that the main [Problem][evotorch.core.Problem] instance remains on the CPU, and resultingly, [SolutionBatch][evotorch.core.SolutionBatch] instances created by [SearchAlgorithm][evotorch.algorithms.searchalgorithm.SearchAlgorithm] instances attached to the problem will also be on the CPU.
 
-This does not, however, mean that you cannot still use CUDA-capable devices for evaluation within the individual Ray actors. To use the GPU, simply use the `num_gpus_per_actor` keyword argument to specify how many GPUs will be allocated to each actor. For example, if you have 2 GPUs, you can do:
+This does not, however, mean that you cannot still use CUDA-capable devices for evaluation within the individual Ray actors.
+When working with a custom [Problem][evotorch.core.Problem] subclass, from within the methods `_evaluate(self, ...)` or `_evaluate_batch(self, ...)`, a GPU device can be obtained (as a `torch.device` instance) via the property `self.aux_device` (where `aux_device` stands for "auxiliary device"). If this custom problem at hand is not parallelised through Ray, the `aux_device` property will return the first visible CUDA-capable device within the main execution environment. If the problem is parallelised through Ray, the `aux_device` property will return the first visible CUDA-capable device within the environment of the actor (therefore, for each actor, it will return the device assigned to that actor). For example, consider this simple custom problem:
 
 ```python
-problem = Problem(
+def vectorised_sphere(x: torch.Tensor) -> torch.Tensor:
+    return torch.sum(x.pow(2.0), dim=-1)
+
+
+class VecSphere(Problem):
+    def _evaluate_batch(self, solutions: SolutionBatch):
+        xs = solutions.values.to(self.aux_device)
+        fs = vectorised_sphere(xs)
+        solutions.set_evals(fs.to(solutions.device))
+```
+
+when a `SolutionBatch` is passed to a `VecSphere` instance's `_evaluate_batch` method, its `values` property is automatically moved to the local `aux_device` for evaluation. Once the decision values have been evaluated (using the `vectorised_sphere` function in this case),  then their fitnesses are moved back to the original device of `solutions` and assigned as the fitness evaluations.
```

Comment:
I think the "automatically" is unnecessarily confusing here :) I'd argue it's not automatic.

Reply:
True, I've removed the word :)